packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/test/Providers/ClientProviders/ClientProviderTests.cs

```diff
@@ -465,24 +466,25 @@ public static IEnumerable<TestCaseData> BuildFieldsTestCases
                 }, false);
                 yield return new TestCaseData(new List<InputParameter>
                 {
+                    // have to explicitly set isRequired because we now call CreateParameter in buildFields
                     InputFactory.Parameter(
-                        "optionalParam",
+                        "optionalNullableParam",
                         InputPrimitiveType.String,
                         location: RequestLocation.None,
                         defaultValue: InputFactory.Constant.String("someValue"),
-                        kind: InputOperationParameterKind.Client),
+                        kind: InputOperationParameterKind.Client, isRequired: false),
```

Comment:
I think we should put the new parameter in a new line, similar as others

Reply:
Also a question: what is the default value of this `isRequired` parameter? Is it `true` or `false`? do we really need to change this?